src/pages/workspace/accounting/AccountingContext.tsx

```diff
@@ -0,0 +1,138 @@
+import type {MutableRefObject, RefObject} from 'react';
+import React, {useMemo, useRef, useState} from 'react';
+import type {View} from 'react-native';
+import AccountingConnectionConfirmationModal from '@components/AccountingConnectionConfirmationModal';
+import useLocalize from '@hooks/useLocalize';
+import {removePolicyConnection} from '@libs/actions/connections';
+import Navigation from '@libs/Navigation/Navigation';
+import {isControlPolicy} from '@libs/PolicyUtils';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type {ConnectionName} from '@src/types/onyx/Policy';
+import type Policy from '@src/types/onyx/Policy';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+import {getAccountingIntegrationData} from './utils';
+
+type ActiveIntegration = {
+    name: ConnectionName;
+    shouldDisconnectIntegrationBeforeConnecting?: boolean;
+    integrationToDisconnect?: ConnectionName;
+};
+
+type ActiveIntegrationState = ActiveIntegration & {key: number};
+
+type AccountingContextType = {
+    activeIntegration?: ActiveIntegration;
+    startIntegrationFlow: (activeIntegration: ActiveIntegration) => void;
+
+    /*
+     * This stores refs to integration buttons, so the PopoverMenu can be positioned correctly
+     */
+    integrationRefs: RefObject<Record<string, MutableRefObject<View | null>>>;
+};
+
+const integrationRefsInitialValue = Object.values(CONST.POLICY.CONNECTIONS.NAME).reduce((acc, key) => {
+    acc[key] = {current: null};
+    return acc;
+}, {} as Record<ConnectionName, MutableRefObject<View | null>>);
+
+const defaultAccountingContext = {
+    activeIntegration: undefined,
+    startIntegrationFlow: () => {},
+    integrationRefs: {
+        current: integrationRefsInitialValue,
+    },
+};
+
+const AccountingContext = React.createContext<AccountingContextType>(defaultAccountingContext);
+
+type AccountingContextProviderProps = ChildrenProps & {
+    policy: Policy;
+};
+
+function AccountingContextProvider({children, policy}: AccountingContextProviderProps) {
+    const integrationRefs = useRef<Record<string, MutableRefObject<View | null>>>(defaultAccountingContext.integrationRefs.current);
```

Comment:
Can you help me understand how we're using `integrationRefs`? I see it initialized with `integrationRefsInitialValue` but I don't see where it's updated.

Reply:
We're changing integration refs in `PolicyAccountingPage` to pass the position of the `threeDotsMenuContainer` button to be able to show modal in the correct place

Reply:
Oh okay I see the name changed to `integrationsRef` in that file. Lets rename that to `popoverAnchorRefs` to be a bit more clear. And let's keep the naming consistent in every file.